climakitae/threshold_tools.py

```diff
@@ -884,3 +889,523 @@ def return_period(ams):
+        """If the button was clicked and the location, variable, or units were 
+        changed, reload the data from AWS"""
+        if self.changed_loc_and_var:
+            self.da = _read_from_catalog(selections = self.selections, location = self.location, cat = self._cat).compute()
+            self.changed_loc_and_var = False
+        if self.changed_units:
+            self.da = _convert_units(da=self.da, selected_units=self.units2)
+            self.threshold_value = round(self.da.mean().values.item())
+            self.param.threshold_value.label = f"Value (units: {self.da.units})"
+            self.changed_units = False
+
+    @param.depends("variable2", watch=True)
+    def _update_variable(self):
+        """Update variable in selections object to reflect variable chosen in panel"""
+        self.selections.variable = self.variable2
+
+    @param.depends("area_subset2", watch=True)
+    def _update_cached_area(self):
+        """
+        Makes the dropdown options for 'cached area' reflect the type of area subsetting
+        selected in 'area_subset' (currently state, county, or watershed boundaries).
+        """
+        if self.area_subset2 == "CA counties":
+            # Get full list of counties
+            self.param["cached_area2"].objects = list(
+                self.location._geography_choose[self.area_subset2].keys()
+            )
+            self.cached_area2 = "Sacramento County"
+        elif self.area_subset2 == "states":
+            # Only CA for now?
+            self.param["cached_area2"].objects = ["CA"]
+            self.cached_area2 = "CA"
+
+    @param.depends("area_subset2","cached_area2",watch=True)
+    def _updated_location(self):
+        """Update locations object to reflect location chosen in panel"""
+        self.location.area_subset = self.area_subset2
+        self.location.cached_area = self.cached_area2
+
+    def transform_data(self):
+        return get_exceedance_count(self.da, 
+            threshold_value = self.threshold_value, 
+            threshold_direction = self.threshold_direction,
+            duration1 = (self.duration1_length, self.duration1_type),
+            period = (self.period_length, self.period_type),
+            groupby = (self.group_length, self.group_type),
+            duration2 = (self.duration2_length, self.duration2_type),
+            smoothing = self.num_timesteps if self.smoothing == "Running mean" else None)
+
+    @param.depends("reload_plot", "reload_data", watch=False)
+    def view(self):
+        try:
+            to_plot = self.transform_data()
+            obj = plot_exceedance_count(to_plot)
+        except Exception as e:
+            # Display any raised Errors (instead of plotting) if any of the 
+            # user specifications are incompatible or not yet implemented.
+            return e
+        return pn.Column(
+            pn.widgets.Button.from_param(self.param.reload_plot, button_type="primary", width=150, height=30),
```

Comment:
I think it would make more sense to just have one reload button, instead of two. The reload_plot button should reload the data, and reload the threshold tools selection. 

In the warming levels panel, we've set up a workflow that only reloads the data if the user has made changes to certain selections related to the data. You could do the same here: have one button (reload plot) that reloads the data ONLY IF the user has made changes to the data selections, and updates the plot with the threshold event options always. 

The code from the warming levels panel that I'm referencing is below. It will reload the postage stamp data when changes to the location subset and variable are made, but NOT if the warming level is changed. If only the warming level is changed, the data (loaded in a previous step, and saved in the variable postage_data) is not reloaded, but the anomalies with that data are recalculated. Thus, avoiding reloading the same data, but allowing for only one button to represent the two changes. 


```
@param.depends("area_subset2","cached_area2","variable2", watch=True)
    def _updated_bool_loc_and_var(self):
        """Update boolean if any changes were made to the location or variable"""
        self.changed_loc_and_var = True

    @param.depends("reload_data2", watch=True)
    def _update_postage_data(self):
        """If the button was clicked and the location or variable was changed,
        reload the postage stamp data from AWS"""
        if self.changed_loc_and_var == True:
            self.postage_data = _get_postage_data(area_subset2=self.area_subset2, cached_area2=self.cached_area2, variable2=self.variable2, location=self.location)
            self.changed_loc_and_var = False
        self._warm_all_anoms = get_anomaly_data(data=self.postage_data, warmlevel=self.warmlevel)
```